### PR TITLE
Cert fix

### PR DESCRIPTION
**Pull request by tmjd**
## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

File: pkg/controller/compliance/compliance_controller.go [hunk 1]
```diff
@@ -442,6 +442,9 @@ func (r *ReconcileCompliance) Reconcile(ctx context.Context, request reconcile.R
 	}
 
 	reqLogger.V(3).Info("rendering components")
+
+	namespaceComp := render.NewPassthrough(render.CreateNamespace(render.ComplianceNamespace, network.KubernetesProvider, render.PSSPrivileged))
```
Comment (caseydavenport) [inline pkg/controller/compliance/compliance_controller.go:446]: I'm not sure I understand the relevance of this change

**Review by caseydavenport — APPROVED** [review]

File: pkg/controller/compliance/compliance_controller.go (hunk 1 above)
Comment (tmjd) [reply pkg/controller/compliance/compliance_controller.go:446]: The certificates need to be created/updated before the compliance deployments are updated and to do that we need the namespace created first.

File: pkg/render/compliance.go [hunk 2]
```diff
@@ -85,6 +87,14 @@ var (
 	ComplianceReporterSourceEntityRule    = networkpolicy.CreateSourceEntityRule(ComplianceNamespace, ComplianceReporterName)
 )
 
+// Register secret/certs that need Server and Client Key usage
+func init() {
+	certkeyusage.SetCertKeyUsage(ComplianceServerCertSecret, []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth, x509.ExtKeyUsageServerAuth})
```
Comment (rene-dekker) [inline pkg/render/compliance.go:92]: When considering the init function, what made you decide to do it in the render file, rather than the controller?

Comment (tmjd) [reply pkg/render/compliance.go:92]: Mainly because that is where the secret name is defined.

**Pull request merged**